public/app/plugins/datasource/stackdriver/datasource.ts

```diff
@@ -109,13 +109,17 @@ export default class StackdriverDatasource {
 
         const unit = this.resolvePanelUnitFromTargets(options.targets);
         queryRes.series.forEach(series => {
-          result.push({
+          let timeSerie = {
             target: series.name,
             datapoints: series.points,
             refId: queryRes.refId,
             meta: queryRes.meta,
             unit,
-          });
+          };
+          if (unit) {
```

Comment:
@sunker Probably me that don't have that good knowledge of javascript, but I find this a bit hard to read/understand. First you insert unit on `timeSerie` above:

```javascript
let timeSerie = {
  target: series.name,
  datapoints: series.points,
  refId: queryRes.refId,
  meta: queryRes.meta,
  unit,
};
```

Then you check if unit is defined and then you set it again in `timeSerie`. 

Shouldn't the first insert of unit to `timeSerie` be removed`?

Reply:
Oh you're absolutely right, my bad. I guess the reason it still works is because unit is undefined (and not 'none' anymore) when returned to the panel. I'll fix it straight away.